### PR TITLE
vengi-tools: init at 0.0.14

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/glm/default.nix [hunk 1]
```diff
@@ -11,6 +16,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-F//+3L5Ozrw6s7t4LrcUmO7sN30ZSESdrPAYX57zgr8=";
   };
 
+  # https://github.com/g-truc/glm/pull/1055
+  # Fix more implicit-int-float-conversion warnings
+  # (https://github.com/g-truc/glm/pull/986 wasn't enough, and -Werror is used)
+  patches = [(fetchpatch {
+    url = "https://github.com/g-truc/glm/commit/bd9b5060bc3b9581090d44f15b4e236566ea86a6.patch";
```
Comment (smancill) [inline pkgs/development/libraries/glm/default.nix:23]: I'd rather use the fork url, to make it clear it's not an upstream patch (yet).

```suggestion
    url = "https://github.com/kraj/glm/commit/bd9b5060bc3b9581090d44f15b4e236566ea86a6.patch";
```

And please send this as its own PR, for #144627

Comment (fgaz) [reply pkgs/development/libraries/glm/default.nix:23]: Will do

**Comment by fgaz** [comment]
Ah sorry, I see ofborg continues to request your review even though it's a draft... I'll remove the glm patch in the next push

**Comment by smancill** [comment]
I had already resigned to my fate.

File: pkgs/development/libraries/glm/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/development/libraries/glm/default.nix]: #147051

**Review by kevincox — APPROVED** [review]

**Comment by grahamc** [comment]
Just a note that the NixOS test associated with this tool has been failing for several months: https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.vengi-tools.x86_64-linux. I wonder if the test is really needed?

**Comment by fgaz** [comment]
@grahamc I'd like to keep it: it makes it easier for me to update the package without having to build/run it on my machine. The OCR is really flaky though, I'll remove that and keep just the screenshot (related: https://github.com/ryantm/nixpkgs-update/issues/298 )

**Comment by fgaz** [comment]
@grahamc looks like I already did that in #190014. It's been sitting there unmerged for a while though...
